### PR TITLE
MGMT-9548: adding API to get DTK info based on DTK image

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
This PR adds an API that allows to get an DTK info (full kernel, rt kernel etc')
encoded inside the DTK image

### CODE REVIEW

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -146,8 +147,22 @@ func (ci *clusterInfo) driverToolkitVersion(ctx context.Context, dtkImages []str
...
+	// we're going only to update the struct with DTK information on
+	// running kernels and not on all that are found.
+	// We could have many entries with DTKs that are from an old update
+	// The objects that are kernel affine should only be replicated
+	// for valid kernels.
+	return ci.updateInfo(info, *dtk, imageURL)
+}
 
-	if dtk = ci.cache[imageURL]; dtk != nil {
+func (ci *clusterInfo) GetDTKData(ctx context.Context, imageURL string) (*registry.DriverToolkitEntry, error) {
```
Comment (pacevedom) [inline pkg/upgrade/upgrade.go:163]: I guess this will be used for the preflight checks as well, right?

Comment (yevgeny-shnaidman) [reply pkg/upgrade/upgrade.go:163]: yes

File: pkg/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -350,3 +352,84 @@ var _ = Describe("ClusterInfo", func() {
 		})
 	})
 })
+
+var _ = Describe("GetDTKData", func() {
```
Comment (pacevedom) [inline pkg/upgrade/upgrade_test.go:356]: Since we now have a dedicated function, could we remove the caching tests for `GetClusterInfo`?

Comment (yevgeny-shnaidman) [reply pkg/upgrade/upgrade_test.go:300]: yes, fixed

**Review by pacevedom — APPROVED** [review]
/lgtm

**Pull request merged**